tests/systemtests/register_supernode_test.go

```diff
@@ -368,23 +401,42 @@ func TestSupernodeRegistrationFailures(t *testing.T) {
 			)
 			t.Logf("Registration response: %s", registerResp)
 
-			// Verify transaction failed with correct error
-			t.Log("Verifying transaction failure")
-			RequireTxFailure(t, registerResp, tc.expectedError)
-
-			// Verify no supernode was registered (except for duplicate_registration case)
-			if tc.name != "duplicate_registration" {
-				t.Log("Verifying no supernode was registered")
-				supernodeResp := cli.WithRunErrorsIgnored().CustomQuery(
-					"query", "supernode", "get-super-node", valAddr,
-				)
-				t.Logf("Supernode query response: %s", supernodeResp)
-
-				require.True(t,
-					strings.Contains(supernodeResp, "not found") ||
-						strings.Contains(supernodeResp, "no supernode found") ||
-						strings.Contains(supernodeResp, "key not found"),
-					"supernode should not be registered, got response: %s", supernodeResp)
+			// Handle re-registration success case
+			if tc.expectedError == "" {
+				t.Log("Verifying transaction success for re-registration")
+				RequireTxSuccess(t, registerResp)
+				
+				// Verify supernode is now active after re-registration
+				sut.AwaitNextBlock(t)
+				supernode := GetSuperNodeResponse(t, cli, valAddr)
+				require.Equal(t, valAddr, supernode.ValidatorAddress)
+				require.NotEmpty(t, supernode.States)
+				lastState := supernode.States[len(supernode.States)-1]
```

Comment:
we can assert the state transition here, not just the last state:
assert len(sn.States) increased by 1 vs. before, and the last two states are Disabled, Active

```
// Verify state transition shape: ... Disabled -> Active
require.GreaterOrEqual(t, len(sn.States), 2)
prev := sn.States[len(sn.States)-2].State
last := sn.States[len(sn.States)-1].State
require.Equal(t, types.SuperNodeStateDisabled, prev)
require.Equal(t, types.SuperNodeStateActive, last)
```